### PR TITLE
Add support for Backblaze B2 Cloud Storage

### PR DESCRIPTION
**Pull request by kamilZ**
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | yes |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | no |
| License | MIT |

---
This change is [Reviewable](https://reviewable.io/reviews/knplabs/gaufrette/357)

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -4,3 +4,5 @@ tests/adapters/*
 !tests/adapters/*.dist
 bin/*
 !bin/configure_test_env.sh
+/.idea
```
Comment (akovalyov) [inline .gitignore]: Could you remove these two lines?
I doubt that open-source libs should take care about particular IDE.

File: composer.json [hunk 2]
```diff
@@ -1,71 +1,72 @@
 {
-    "name":         "knplabs/gaufrette",
```
Comment (akovalyov) [inline composer.json]: Could you revert the formatting for better readability?

File: src/Gaufrette/Adapter/BackblazeB2Storage.php [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+            return false;
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isDirectory($key)
+    {
+        if ($this->exists($key . '/')) {
```
Comment (akovalyov) [inline src/Gaufrette/Adapter/BackblazeB2Storage.php]: I am not familiar with `Backblaze`, but what happens if I pass `/foo/bar/` directory as parameter here?

File: .gitignore (hunk 1 above)
Comment (kamilZ) [inline .gitignore]: Ok

File: src/Gaufrette/Adapter/BackblazeB2Storage.php (hunk 3 above)
Comment (kamilZ) [inline src/Gaufrette/Adapter/BackblazeB2Storage.php]: Always we'll get bad request: "file names must not end with '/'". 
There is no way to check if "/foo/bar/" exist in B2 atm.

I'll change this to return false always.

File: composer.json (hunk 2 above)
Comment (kamilZ) [inline composer.json]: sorry for that.

**Comment by NiR-** [comment]
Sorry @kamilZ for the really late response.

Unfortunately, we'll not merge your new adapter because we want to reduce the number of them we need to maintain. But, in the case you release your adapter as an independant FOSS project, I would be really happy to add a link in our readme/docs to your adapter :)

**Comment by AntoineLelaisant** [comment]
Could we close this PR then @NiR- ?

**Pull request closed without merging**